lib/shopify-cli/context.rb

```diff
@@ -10,6 +10,44 @@ module ShopifyCli
   # resoures.
   #
   class Context
+    class << self
+      attr_reader :messages
+
+      # adds a new set of messages to be used by the CLI from the given YAML file. This will flatten the keys using '.'s
+      # and will fail if any of the new keys conflict with existing ones.
+      #
+      # #### Parameters
+      # * `yaml_file_path` - The path of a YAML file containing the messages
+      def load_messages_file(yaml_file_path)
+        require 'yaml'
+
+        yaml_contents = YAML.load_file(yaml_file_path)
+        return unless yaml_contents
+
+        @messages = {} if @messages.nil?
+        @messages = @messages.merge(flatten_yaml_file(yaml_contents)) do |key|
+          abort("Message key '#{key}' already exists and cannot be registered") if @messages.key?(key)
+        end
+      end
+
+      private
+
+      def flatten_yaml_file(yaml_data)
+        messages = {}
+        yaml_data.each do |key, value|
+          if value.is_a?(Hash)
+            flatten_yaml_file(value).each do |inner_key, inner_value|
+              messages["#{key}.#{inner_key}"] = inner_value
```

Comment:
I get what you are doing here but you don't need to do this!

In the `message` method you could just use `dig`

```ruby
    def message(key, *params)
      str = @messages.dig(key.split('.'))
      str ? str % params : key
    end
```